backend/api/pipeline.proto

```diff
@@ -209,6 +216,14 @@ message ListPipelinesResponse {
   string next_page_token = 2;
 }
 
+message GetPipelineByNameRequest {
+  // The Name of the pipeline to be retrieved.
+  string name = 1;
+  // For Pipeline, the only valid resource type is Namespace. An sample query string could be
+  // resource_reference.type=NAMESPACE&resource_reference.id=ns1
+  ResourceKey resource_reference_key = 2;
```

Comment:
Thank you for working on this!

nit: I have some different personal opinion about this.

I understand that the current convention is to use resource ref to filter resources. However, I think it's worth reconsidering that.

For example, to use this API, user needs to understand the complex resource reference key concept. Furthermore, they need to read the comment above and understand they can only filter by namespace here.

What if we just add an optional namespace field here? It's much much easier to understand and use.

The argument also applies to other endpoints, but because this is a new endpoint, we might consider following a different approach if we all agree with it.

Reply:
I expanded the topic into an issue: https://github.com/kubeflow/pipelines/issues/7169